### PR TITLE
add support for cinder backups

### PR DESCRIPTION
**Pull request by tzumainn**

### CODE REVIEW

File: lib/fog/volume/openstack/requests/list_backups_detailed.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            :expects  => 200,
+            :method   => 'GET',
+            :path     => 'backups/detail',
+            :query    => options
+          )
+        end
+      end
+
+      module Mock
+        def list_backups_detailed(options = {})
```
Comment (gildub) [inline lib/fog/volume/openstack/requests/list_backups_detailed.rb]: Just use _options should do it.

File: lib/fog/volume/openstack/requests/get_backup_details.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          request(
+            :expects  => 200,
+            :method   => 'GET',
+            :path     => "backups/#{backup_id}"
+          )
+        end
+      end
+
+      module Mock
+        def get_backup_details(backup_id)
```
Comment (gildub) [inline lib/fog/volume/openstack/requests/get_backup_details.rb]: Same as above, use `_backup_id`.

File: lib/fog/volume/openstack/requests/delete_backup.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+          request(
+            :expects => 202,
+            :method  => 'DELETE',
+            :path    => "backups/#{backup_id}"
+          )
+        end
+      end
+
+      module Mock
+        def delete_backup(backup_id)
```
Comment (gildub) [inline lib/fog/volume/openstack/requests/delete_backup.rb]: Same same

File: lib/fog/volume/openstack/requests/create_backup.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            :body => Fog::JSON.encode(data),
+            :expects => [200, 202],
+            :method => 'POST',
+            :path => 'backups'
+          )
+        end
+      end
+
+      module Mock
+        def create_backup(options = {})
```
Comment (gildub) [inline lib/fog/volume/openstack/requests/create_backup.rb]: Idem

File: lib/fog/volume/openstack/models/backup.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+        def restore(volume_id)
+          requires :id
+          service.restore_backup(id, volume_id)
+          true
+        end
+
+        def volume
+          requires :id
+          service.get_volume_details(self.volume_id).body['volume']
```
Comment (gildub) [inline lib/fog/volume/openstack/models/backup.rb]: You can do `service.get_volume_details(volume_id).body['volume']`

**Comment by gildub** [comment]
LGTM, especially knowing that the tests are on their way :)
What about adding examples to docs/ and /examples ?

**Comment by tzumainn** [comment]
@gildub I've added examples (docs seem to be missing for volumes as a whole) - let me know what you think!

**Comment by Ladas** [comment]
@tzumainn looks great, thank you for the contribution :-)

**Pull request merged**